spec.md

```diff
@@ -231,15 +228,36 @@ extmodule MyExternalModule :
   input foo: UInt<2>
   output bar: UInt<4>
   output baz: SInt<8>
-  defname = VerilogName
-  parameter x = "hello"
-  parameter y = 42
 ```
 
 The widths of all externally defined module ports must be specified.  Width
 inference, described in [@sec:width-inference], is not supported for module
 ports.
 
+Externally defined modules may have zero or more parameters.  Parameters may be
+of known-width `UInt`{.firrtl} or `SInt`{.firrtl} types or `String`{.firrtl}
+type.  The value of a parameter is set at each instantiation of an external
+module using a literal value.
+
+An example of a parametric externally defined module and its instantiation is:
+
+``` firrtl
+extmodule MyExternalModule2<
+  parameter x: String,
+  parameter y: UInt<8>,
+  parameter z: SInt<4>
+> :
+  ; ...
```

Comment:
Not sure what this semicolon is doing here? It's a little confusing. Can we just show the full module with some ports?

Also is the expectation that those parameters could define the port width (this seems like a common use case), can you show that usage?